packages/sdk/src/mint/mint.ts

```diff
@@ -36,75 +35,91 @@ export const mint = (
     throw new Error(ERROR_MESSAGES.NO_MEDIA);
   }
 
-  if (splits) {
+  let adjustedRoyalties: Splits = {};
+  let royaltyTotal: number;
+
+  if (royalties) {
     // FIXME: suggest we use % (ints) vs float here
     // 0.5 -> 5000
-    adjustSplitsForContract(splits);
+    royaltyTotal = getRoyaltyTotal(royalties);
+    adjustedRoyalties = adjustRoyaltiesForContract(royalties, royaltyTotal);
   }
 
-  if (splits && Object.keys(splits).length > 50) {
-    throw new Error(ERROR_MESSAGES.MAX_SPLITS);
-  }
-
-  if (splits && Object.keys(splits).length < 2) {
-    throw new Error(ERROR_MESSAGES.SPLITS);
+  if (royalties && Object.keys(royalties).length < 1) {
+    throw new Error(ERROR_MESSAGES.MIN_ROYALTIES);
   }
 
   if (amount && amount > 125) {
     throw  new Error(ERROR_MESSAGES.MAX_AMOUT);
   }
-
-  if (royaltyPercentage && royaltyPercentage < 0 || royaltyPercentage > 0.5) {
-    throw new Error(ERROR_MESSAGES.INVALID_ROYALTY_PERCENTAGE);
+  if (tokenIdsToMint && tokenIdsToMint.length == 0) {
+    throw  new Error(ERROR_MESSAGES.MAX_AMOUT);
   }
 
+  const adjustedTokenCount = tokenIdsToMint?.length > amount? tokenIdsToMint.length : amount;
```

Comment:
This is more headaches. So we can mint 20, and maybe I only want to specify 5 token IDs, but it's still valid?

Reply:
Throws error now